src/components/profile/VerifyEdit.js

```diff
@@ -56,9 +56,8 @@ const EditProfile = ({ screenProps, theme, styles, navigation }) => {
       if (data.alreadyVerified) {
         await userStorage.setProfileField(field, content)
         screenProps.pop()
-      } else {
-        screenProps.push('VerifyEditCode', { field, content })
       }
+      screenProps.push('VerifyEditCode', { field, content })
```

Comment:
if we are neways calling .push no matter the response from the API, why do we need the `if(data.alreadyVerified)`?

Reply:
+1

Also why do we need to redirect onto VerifyEditCode and send code again if already verified ?
The solution looks like some kind of the hack/workaround, could you please provide detailed explanations ?

Reply:
we need to verify because this is the case where the user edited his phone number, so we need to verify the new number

> +1
> 
> Also why do we need to redirect onto VerifyEditCode and send code again if already verified ?
> The solution looks like some kind of the hack/workaround, could you provide detailed explanations please ?

Reply:
@omerzam 

1. why do we need the following block
```
 if (data.alreadyVerified) {        
   await userStorage.setProfileField(field, content)       
   screenProps.pop()      
}
```

if we'll send/check the code anyway, it means we shouldn't update profile right now, we have to do it only after new phone has been confirmed

2. do we checked before was the phone number changed or not ? Won't this change to force SMS code on any profile update ? E.g. would we need to confirm code if we changed other profile field like username ?